### PR TITLE
fix(table): auto-collapse expanded row upon clicking another expandable row

### PR DESCRIPTION
**Pull request by ckrook**
## **Describe pull-request**  
This pull-request adds the possibility for the developer to enable "auto collapse". This means that if its enabled, all other rows except for the row selected will be closed.

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** [CDEP-3490](https://tegel.atlassian.net/browse/CDEP-3490)
- **GitHub:**  [Issue 379](https://github.com/scania-digital-design-system/tegel/issues/739)

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Go to Storybook
2. Navigate to the table component 
3. Go to expandable rows
4. Toggle the control autoCollapse to true
5. Test open multiple rows now, the expected behavior is that all other rows expect the selected row should close

## **Checklist before submission**
- [x] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [x] I have updated the documentation (if applicable)

## **Suggested test steps**
- [x] Browser testing (Chrome, Safari, Firefox) 
- [x] Keyboard operability
- [x] Storybook controls

[CDEP-3490]: https://tegel.atlassian.net/browse/CDEP-3490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/components/table/table-component-expandable-rows.stories.tsx [hunk 1]
```diff
@@ -131,20 +131,41 @@ export default {
...
+      verticalDivider: false,
+      noMinWidth: false,
+      column1Width: '',
+      column2Width: '',
+      column3Width: '',
+      column4Width: '',
+      overflow: 'scroll',
+      autoCollapse: false,
+    },
+    autoCollapse: {
```
Comment (nathalielindqvist) [inline packages/core/src/components/table/table-component-expandable-rows.stories.tsx:159]: Seems like there might be a duplicate of this Storybook control, it takes in "args" as a control and doesn't set a default value for the "autoCollapse" control
Screenshot 2024-10-11 at 09 24 02

**Comment by theJohnnyMe** [comment]
Something is off with the story controls on this one, lets take a look together on it later today. 
Screenshot 2024-10-16 at 10 33 18

**Comment by theJohnnyMe** [comment]
I did rebase with the latest develop branch. The test related to this is failing.

**Review by timrombergjakobsson — APPROVED** [review]

**Review by Lunkan89 — APPROVED** [review]

**Review by timrombergjakobsson — APPROVED** [review]

**Comment by ckrook** [comment]
Last Thursday before release Nathalie noticed in Storybook that each row was logged upon click on a row.
I have resolved this issue with my latest commit.

**Review by nathalielindqvist — COMMENTED** [review]
The preview link is still showing the previous issues with the multiple id's, the build might need quite som time before updating 🤔 
However, when running the branch locally I get an error when setting Auto Collapse to true:
Screenshot 2024-10-22 at 10 41 18

No id's are being logged in the console after that

**Comment by ckrook** [comment]
@nathalielindqvist Thanks for the catch, I have committed a fix for this :)

File: packages/core/src/components/table/table-component-expandable-rows.stories.tsx [hunk 2]
```diff
@@ -130,34 +130,32 @@ export default {
       control: {
         type: 'radio',
       },
-
       options: ['auto', 'hidden'],
```
Comment (nathalielindqvist) [inline packages/core/src/components/table/table-component-expandable-rows.stories.tsx:133]: The default value to Overflow seems to have gotten lost along the way, could we maybe set that to "auto"?
Screenshot 2024-10-24 at 13 08 34

File: packages/core/src/components/table/table-component-expandable-rows.stories.tsx [hunk 3]
```diff
@@ -130,34 +136,35 @@ export default {
...
+    compactDesign: false,
+    expanded: false,
+    responsiveDesign: false,
+    verticalDivider: false,
+    noMinWidth: false,
+    column1Width: '',
+    column2Width: '',
+    column3Width: '',
+    column4Width: '',
+    overflow: 'scroll',
```
Comment (nathalielindqvist) [inline packages/core/src/components/table/table-component-expandable-rows.stories.tsx:166]:
```suggestion
    overflow: 'auto',
```

File: packages/core/src/components/table/table-component-expandable-rows.stories.tsx (hunk 2 above)
Comment (ckrook) [reply packages/core/src/components/table/table-component-expandable-rows.stories.tsx]: This, but also there should be three values: 'auto' | 'hidden' | 'visible' :)

**Review by nathalielindqvist — APPROVED** [review]
Nice! Please be mindful when merging if [this PR](https://github.com/scania-digital-design-system/tegel/pull/811) is merged before yours, you might need to rebase again in that case :)

**Pull request merged**